app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -136,7 +154,7 @@ private TensorFlow getInitializedTensorFlow(){
         return tensorFlow;
     }
 
-    private boolean isStudentImageValid(StudentImage studentImage){
+    private synchronized boolean isStudentImageValid(StudentImage studentImage){
```

Comment:
What does "valid" mean? Can you add some documentation to this method explaining this?